### PR TITLE
[MAINTENANCE] Clean up Metrics type usage, SQLAlchemyExecutionEngine and SQLAlchemyBatchData implementation, and SQLAlchemy API usage

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-454
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/execution_engine/sqlalchemy_batch_data.py [hunk 1]
```diff
@@ -240,6 +242,13 @@ def _create_temporary_table(
             stmt = "CREATE TABLE {temp_table_name} AS {query}".format(
                 temp_table_name=temp_table_name, query=query
             )
+        elif self.sql_engine_dialect.name.lower() == "trino":
+            if temp_table_schema_name is not None:
+                temp_table_name = temp_table_schema_name + "." + temp_table_name
+
+            stmt = "CREATE MATERIALIZED VIEW {temp_table_name} AS {query}".format(
+                temp_table_name=temp_table_name, query=query
+            )
```
Comment (Shinnnyshinshin) [inline great_expectations/execution_engine/sqlalchemy_batch_data.py]: oo new dialect :)

Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_batch_data.py]: Deleted -- not yet!  Nice catch. :)

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -898,7 +899,12 @@ def build_pandas_validator_with_data(
     batch_definition: Optional[BatchDefinition] = None,
 ) -> Validator:
     batch: Batch = Batch(data=df, batch_definition=batch_definition)
-    return Validator(execution_engine=PandasExecutionEngine(), batches=(batch,))
```
Comment (Shinnnyshinshin) [inline great_expectations/self_check/util.py:901]: Thank you for catching this. I see that the contructor for `Validator` takes in `batches: Optional[List[Batch]]`.  Have we "always" (relatively-speaking) been sending in a Tuple here?

Comment (alexsherstinsky) [reply great_expectations/self_check/util.py:901]: Quite often -- I fixed a number of them before; now saw this one too.

Comment (Shinnnyshinshin) [reply great_expectations/self_check/util.py:901]: I see.. thank you for catching this @alexsherstinsky

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great :) Thanks for the clean up

**Pull request merged**